krun/cpu.py

```diff
@@ -0,0 +1,117 @@
+# Various CPU related abstrations
+
+import time
+import os
+from collections import OrderedDict
+from krun.util import fatal, collect_cmd_output
+
+
+class BasePlatform(object):
+    CPU_TEMP_MANDATORY_WAIT = 1
+    CPU_TEMP_POLL_FREQ = 5
+
+    def __init__(self):
+        self.audit = OrderedDict()
+
+    def wait_until_cpu_cool(self):
+        time.sleep(BasePlatform.CPU_TEMP_MANDATORY_WAIT)
+        msg_shown = False
+        while True:
+            cool, reason = self.has_cpu_cooled()
+            if cool:
+                break
+
+            # if we get here, CPU is too hot!
+            if not msg_shown:
+                print("CPU is running hot.")
```

Comment:
I don't see a problem with the CPU being temporarily hot -- presumably the problem is if it's in meltdown mode and/or after several seconds (30? 60? dunno...) it hasn't cooled down? We might want to be careful about scaring the user until we're really sure something's gone wrong.